[Infra] Add new metrics tab and redesign of hosts overview metrics section

crespocarlos opened the pull request:
closes [#175446](https://github.com/elastic/kibana/issues/175446)
## Summary

Adds new Metrics tab and redesigns the metrics section in the Overview tab



https://github.com/elastic/kibana/assets/2767137/766305ce-23c6-4001-ab20-2230decc02a1




### Extra changes
 - Replaced usages of asset name in favor of asset id to filter asset data across the Asset Details components - which will help when it starts supporting more asset types
 - Renamed some translation keys to use `assetDetails` instead of `nodeDetails`

### How to test
- Start a local Kibana instance
- Navigate to Infrastructure > Hosts
- Open the flyout for a host
  - Overview tab 
    - Check if the metrics section matches the design and AC
  - Metrics tab
    - Check if it matches the design the design and AC
- Check the above in the full page view.

crespocarlos committed:
Add new metrics tab and redesign of hosts overview metrics section

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

crespocarlos committed:
Add tests

crespocarlos commented:
/ci

crespocarlos committed:
Merge branch 'main' of github.com:elastic/kibana into 175446-new-metrics-tab

crespocarlos commented:
/ci

crespocarlos committed:
Translations fix

crespocarlos committed:
Clean up

crespocarlos committed:
Use asset.id instead of asset.name

crespocarlos commented:
/ci

crespocarlos committed:
Clean up

crespocarlos committed:
Fix tests

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos committed:
Fix log tooltip external link

crespocarlos commented:
/ci

crespocarlos committed:
Merge branch 'main' of github.com:elastic/kibana into 175446-new-metrics-tab

crespocarlos commented:
/ci

crespocarlos committed:
Fix serverless tests

crespocarlos commented:
/ci

crespocarlos committed:
Adjust Kubernetes section in overview tab

crespocarlos commented:
/ci

crespocarlos committed:
Metrics template improvements

crespocarlos commented:
/ci

crespocarlos commented:
/ci

crespocarlos committed:
Merge branch 'main' of github.com:elastic/kibana into 175446-new-metrics-tab

crespocarlos committed:
Fix after merge

crespocarlos commented:
/ci

crespocarlos committed:
Fix disk IOPS and Throughput charts and tests

crespocarlos commented:
/ci

crespocarlos commented on x-pack/plugins/metrics_data_access/common/inventory_models/host/metrics/charts/disk.ts:
```diff
@@ -16,8 +16,8 @@ import {
 
-const diskThroughputReadWrite: LensConfigWithId = {
-  id: 'diskThroughputReadWrite',
+const diskIOReadWrite: LensConfigWithId = {
```
ids were swapped, but the label and content are correct

elasticmachine commented:
Pinging @elastic/obs-ux-infra_services-team (Team:obs-ux-infra_services)

cauemarcondes committed:
Merge branch 'main' into 175446-new-metrics-tab

jennypavlova commented on x-pack/plugins/observability_solution/infra/public/components/asset_details/translations.ts:
```diff
@@ -10,3 +10,24 @@ import { i18n } from '@kbn/i18n';
   defaultMessage: 'N/A',
 });
+
+export const HOST_METRIC_GROUP_TITLES: Record<string, string> = {
```
Nit: We can make the type stricter - something like: 

```
type HostMetricGroup = 'kubernetes' | Exclude<HostMetricTypes, 'kpi'>;

export const HOST_METRIC_GROUP_TITLES: Record<HostMetricGroup, string> = { ...
```

crespocarlos replied:
Good idea. we can let ts infer the type in this case.

jennypavlova reviewed (commented):
Well done 👏  The metrics tab works as expected 🙌 There are some small things I found I will list here but overall it looks great (and I added some code comments/suggestions):
- The tooltip icon second click is scrolling up (when there is already an open tooltip - maybe an edge case tho)

https://github.com/elastic/kibana/assets/14139027/622c606b-5354-4a72-9bf5-434dd7a1d8fd

- The Docs title and link names are inconsistent (not sure which one should be changed or if we should change it at all but all other links/titles are the same): 
  - CPU -> CPU Usage in the Docs

  https://github.com/elastic/kibana/assets/14139027/6761e95a-f343-47c0-9500-2a3caff8bcb7


  - Logs rate -> Log in the Docs

  https://github.com/elastic/kibana/assets/14139027/915d6ec9-fbd1-444d-a50d-ee4dd3c30fc2


- Nit: The Left/Top Menu alignment looks a bit off - not super visible tho 😅  ( I assume it will look better if the titles and the menu are aligned but up to you to decide of course): 
<img width="1870" alt="image" src="https://github.com/elastic/kibana/assets/14139027/302a0057-f98f-4e44-a205-ff49346fc946">

_Update:_ The icon I mean here ^ I am not sure if it's clear from the screenshot

<img width="581" alt="image" src="https://github.com/elastic/kibana/assets/14139027/ade6f205-4f8f-4490-acc3-19a1baae5416">

- In the[ figma design prototype](https://www.figma.com/proto/8nuZZg3waJhNjHjNhu7rKi/Metrics-Tab-for-Hosts?page-id=0%3A1&type=design&node-id=120-108330&t=K9EvEORUrT5hAm3B-0&scaling=min-zoom) the header is fixed even if the metric needs scrolling:


https://github.com/elastic/kibana/assets/14139027/4ff5ef7c-a717-414e-80c1-6e733870241d



Do we want to keep the header with the tabs in view if we navigate to a specific metric?

crespocarlos commented:
Hey @jennypavlova , thanks for reviewing it. I'm having troubles to watch the videos.

> CPU -> CPU Usage in the Docs
> Logs rate -> Log in the Docs

IMO this is fine and won't cause confusion. The UI link doesn't necessarily need to follow the doc header. The important thing is that it points the user to the right place. I'll defer that to @roshan-elastic .

> Nit: The Left/Top Menu alignment looks a bit off - not super visible tho 😅 ( I assume it will look better if the titles and the menu are aligned but up to you to decide of course)

Fixed!

><img src="https://github.com/elastic/kibana/assets/2767137/b23be7b7-baec-4364-8aa9-64b1fe5f642d" width="300px" />

Fixed.

> 
<img width="600px" alt="image" src="https://github.com/elastic/kibana/assets/2767137/8dca8dfa-8fe1-4c48-8ecf-4ef0546a6f50">

Good catch. Fixed

>  <img width="600px" alt="image" src="https://github.com/elastic/kibana/assets/2767137/52690358-cbcb-4d2c-b88f-b17e5442a07d">

It's a bit hard to make this pixel perfect because there is some calculation involved to align header + side panel item. Especially when automatic scroll happens. I tried to improve the alignment, though.

> In the[ figma design prototype](https://www.figma.com/proto/8nuZZg3waJhNjHjNhu7rKi/Metrics-Tab-for-Hosts?page-id=0%3A1&type=design&node-id=120-108330&t=K9EvEORUrT5hAm3B-0&scaling=min-zoom) the header is fixed even if the metric needs scrolling:
> Do we want to keep the header with the tabs in view if we navigate to a specific metric?

True. Thanks for catching this. Let's do this in a follow-up ticket.

crespocarlos committed:
CR fixes

crespocarlos committed:
Fix quick access items gap

crespocarlos commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 175446-new-metrics-tab

crespocarlos commented:
Hey @jennypavlova , Finally was able to watch the videos

> The tooltip icon second click is scrolling up (when there is already an open tooltip - maybe an edge case tho)

This happens across Kibana, at least in the observability solution. I think it's worth investigating it, but it looks like a broader topic. 

<img width="600" alt="image" src="https://github.com/elastic/kibana/assets/2767137/33bb39e6-9b89-4caa-bf88-a25377290040">


<img width="600" alt="image" src="https://github.com/elastic/kibana/assets/2767137/6499cc15-e2e2-4706-819f-6c58802b595f">

crespocarlos commented:
> The tooltip icon second click is scrolling up (when there is already an open tooltip - maybe an edge case tho)

@jennypavlova This is a problem with popovers within the flyout. it happens with any popover when they get closed and their corresponding button are not visible.  For some reason that I have not found yet, the flyout will automatically scroll to the button that originated the popover. Best to investigate this in a separate ticket.

jennypavlova commented:
> This is a problem with popovers within the flyout. it happens with any popover when they get closed and their corresponding button are not visible. For some reason that I have not found yet, the flyout will automatically scroll to the button that made the popover appear. Best to investigate this in a separate ticket.

@crespocarlos Thank you for investigating the issue! Yes, let's open a new ticket for that. I found it here but I can imagine that this is happening in other places as well. It looks like an unrelated issue but I decided to mention it anyway. I will pull all new changes and retest everything again.

jennypavlova committed:
Merge branch 'main' into 175446-new-metrics-tab

roshan-elastic commented:
@jennypavlova 

> CPU -> CPU Usage in the Docs
> Logs rate -> Log in the Docs
> 
> IMO this is fine and won't cause confusion. The UI link doesn't necessarily need to follow the doc header. The important thing is that it points the user to the right place. I'll defer that to @roshan-elastic .

Yeah - ideally we should update this in the docs. I think it's not a priority...I think the fact we explicitly point them to the right information is good enough. Good catch though...

crespocarlos commented:
> Thank you for investigating the issue! Yes, let's open a new ticket for that. I found it here but I can imagine that this is happening in other places as well. It looks like an unrelated issue but I decided to mention it anyway. I will pull all new changes and retest everything again.

@jennypavlova I found a solution :).

We might want to create a ticket to prevent the popup and popover from staying on top of header and search bar.

crespocarlos committed:
Prevent popover from forcing flyout to scroll

jennypavlova reviewed (approved):
LGTM! 💯  Thank you for adding all the fixes! The fixed header with the tabs in the flyout is really nice 🎉  It will be tricky to have it on the details page, right? 
it will be a nice improvement IMO, it can be a separate issue (it is also present in the videos)
A really small thing I think would be nice to have - when navigating from a section in overview (let's say memory) on the metrics tab the section title in the left menu (memory in our example) is not underlined but it is underlined if it is clicked directly. Do we want to actually underline the section also when we navigate from the overview? I am adding some videos as well for context (please let me know if you can't access them) - it's not a must-have change.

https://github.com/elastic/kibana/assets/14139027/1ce092d3-1beb-4ded-ac0e-ad66818b7add

https://github.com/elastic/kibana/assets/14139027/831114e5-6934-4c12-a92e-79c5fe28da7e

NIT and not blocking: Another design thing we can check is if we want to add an underline under the top menu in case of a smaller screen (I like that now it's aligned but the last menu item is on the next line for a really small screen size and something to show that the menu is separated from the content can be a nice addition) For most cases it's a single line menu and looks good so nothing we should consider now I think:
<img width="884" alt="image" src="https://github.com/elastic/kibana/assets/14139027/ea7aa7b4-5894-4084-9644-859d21c22903">

crespocarlos committed:
Small UI improvements

crespocarlos commented:
@jennypavlova 

Agree with both of your suggestions. Added an underline to the active item and a border below the quick access menu - when on smaller screens.

> The fixed header with the tabs in the flyout is really nice 🎉 It will be tricky to have it on the details page, right?

On the flyout the sticky header is given for free. On the page view, we probably need to reach out to EUI to see if they can provide something for the [template](https://eui.elastic.co/#/templates/page-template/examples) component.

jennypavlova commented:
> Agree with both of your suggestions. Added an underline to the active item and a border below the quick access menu - when on smaller screens.

@crespocarlos Thank you 😊 Both changes look good!

> On the flyout the sticky header is given for free. On the page view, we probably need to reach out to EUI to see if they can provide something for the [template](https://eui.elastic.co/#/templates/page-template/examples) component.

Thanks for investigating! Checking with EUI sounds like a good option 👍

roshan-elastic commented:
Wow wow wow @jennypavlova @crespocarlos - this looks SUPER useful! Can't wait to get this out in front of users!

FYI @kkurstak @sileschristian  (see [video demos](https://github.com/elastic/kibana/pull/180447#pullrequestreview-2000721325) above)
